test/e2e/resize_nodes.go

```diff
@@ -223,6 +257,59 @@ func waitForPodsCreatedRunningResponding(c *client.Client, ns, name string, repl
 	return nil
 }
 
+// Blocks outgoing network traffic on 'node'. Then verifies that 'podNameToDisappear',
+// that belongs to replication controller 'rcName', really disappeared.
+// Finally, it checks that the replication controller recreates the
+// pods on another node and that now the number of replicas is equal 'replicas'.
+// At the end (even in case of errors), the network traffic is brought back to normal.
+// This function executes commands on a node so it will work only for some
+// environments.
+func performTemporaryNetworkFailure(c *client.Client, ns, rcName string, replicas int, podNameToDisappear string, node *api.Node) {
+	Logf("Getting external IP address for %s", node.Name)
+	host := ""
+	for _, a := range node.Status.Addresses {
+		if a.Type == api.NodeExternalIP {
+			host = a.Address + ":22"
+			break
+		}
+	}
+	By(fmt.Sprintf("block network traffic from node %s to the master", node.Name))
+
+	defer func() {
+		By(fmt.Sprintf("Unblock network traffic from node %s to the master", node.Name))
+		// delete OUTPUT rule number 1
+		undropCmd := "sudo iptables --delete OUTPUT 1"
+		if _, _, code, err := SSH(undropCmd, host, testContext.Provider); code != 0 || err != nil {
+			Failf("Expected 0 exit code and nil error when running %s on %s, got %d and %v",
+				undropCmd, node, code, err)
+		}
+	}()
+
+	// The command will block all outgoing network traffic from the node to the master
+	// When multi-master is implemented, this test will have to be improved to block
+	// network traffic to all masters.
+	// We could also block network traffic from the master(s)s to this node,
+	// but blocking it one way is sufficient for this test.
+	// (insert output rule number 1)
+	dropCmd := fmt.Sprintf("sudo iptables -I OUTPUT 1 -d %s -j DROP", testContext.CloudConfig.MasterName)
+	if _, _, code, err := SSH(dropCmd, host, testContext.Provider); code != 0 || err != nil {
+		Failf("Expected 0 exit code and nil error when running %s on %s, got %d and %v",
```

Comment:
I checked this up, and I was wrong.  Fail causes a panic, which causes deferred to be called as usual.  So you're all good.

"Fail (and therefore Gomega, since it uses fail) will record a failure for the current space and panic. This allows Ginkgo to stop the current spec in its tracks - no subsequent assertions (or any code for that matter) will be called. Ordinarily Ginkgo will rescue this panic itself and move on to the next test."